src/main/java/com/checkmarx/flow/CxFlowRunner.java

```diff
@@ -504,16 +524,17 @@ private void cxBatch(ScanRequest request) throws ExitThrowable {
 
     private void cxResults(ScanRequest request) throws ExitThrowable {
         ScanResults results = resultsService.cxGetResults(request, null).join();
-        if(flowProperties.isBreakBuild() && resultsService.filteredIssuesPresent(results)){
+        if ((flowProperties.isBreakBuild() || breakBuildArgument) && resultsService.filteredIssuesPresent(results)){
```

Comment:
isn't breakBuildArgument an override of flowProperties.isBreakBuild().
it should get its value at the start of the request and be used instead of the flowproperties not as a complement to it.